Enable passing test mode False as kwarg from orch file to minion in permanent test mode

Tanoti opened the pull request:
Partial solution for #24209.

### What does this PR do?
Enables passing test mode=False as kwarg from orchestration file down to minion so as to override a minion with test:True enabled in the config file.

### What issues does this PR fix or reference?
#24209

### Previous Behavior
Minion would ignore the test mode override and execute states in test mode regardless.

### New Behavior
Minion now receives and uses the test=False override from the orchestration file.

### Tests written?
No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

Commit:
Add priority field to support the latest capirca.

Commit:
Add priority field to support the latest capirca.

Tanoti committed:
Enable passing test mode as kwarg from orchestration file down to minion so as to override a minion with test:True enabled in the config file.
Partial solution for #24209.

rallytime commented:
@Tanoti This has a lint failure: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/18019/violations/file/salt/states/saltmod.py/ Can you fix it?

Tanoti committed:
Fixed lint failure

rallytime reviewed (changes requested):
This change needs to be documented. Both in the module docs in this file, and probably also might be good to add to the feature release documentation for `Fluorine`.

Tanoti commented:
@rallytime, do I need any of these documentation tasks?

rallytime commented:
@Tanoti Yes, please. If you could. The Fluorine release notes don't have anything in them yet, but you can find them here: https://github.com/saltstack/salt/blob/develop/doc/topics/releases/fluorine.rst

I think in the past we've added a header with something like `State Module Changes` or something along those lines. You can check out the `oxygen` release notes for an example.

Tanoti committed:
Added documentation for saltmod.state test mode

rallytime commented on doc/topics/releases/fluorine.rst:
```diff
@@ -3,3 +3,21 @@
+argument here is left as None then a value of `test=True` on the command-line is
+passed correctly to the minion to run an orchestration in test mode. At present
+it is not possible to pass `test=False` on the command-line to override a
+minion in permanent test mode and so the `test:False` option must stilil be set
```
`still` :)

Tanoti replied:
@rallytime Well, you try and invent a new word and no-one likes it ;)

rallytime commented on doc/topics/releases/fluorine.rst:
```diff
@@ -3,3 +3,21 @@
+states.saltmod
+--------------
+The 'test' option now defaults to None. A value of True or False set here is 
+passed to the state being run and can be used to override a `test:True` option
```
Tip: RST syntax needs two back-ticks if you want `test: True` to show up as code markdown.

rallytime reviewed (changes requested):
Thanks for adding docs @Tanoti - I just had a couple of small comments.

Tanoti committed:
Fixed typo and updated formatting

twangboy committed:
Add utils\mac_service.py

twangboy committed:
Move some functions into mac_utils

twangboy committed:
Fix args/kwargs

twangboy committed:
Fix unicode_literal issue in mac_assistive

twangboy committed:
Used *args and **kwargs

twangboy committed:
Add tests for mac_utils

Need help with the `available_services` tests

twangboy committed:
Add more tests for available_services

twangboy committed:
Add assert_called_with

cachedout commented:
I do not AT ALL think this is a good idea. Why would we allow this to be overwritten? Could you explain your reasoning for this proposal, please?

Tanoti commented:
If you are running minions in "test: True" mode via their config file to prevent unintended operations from users forgetting to set the test mode when making a command line call, then the expected method to override this is to pass "test=false" to say you really want this to happen.

In orchestration this is not possible as a "test" flag on the CLI is not recognised by the minion. This PR is a partial solution to allow us to say in an orchestration file that we really do want to override the minion's test mode. I understand that this may not be the best solution but it is all I could think of while digging through the code. If you are going to reject this then please make "test=false" specified on a salt-run call be passed to, and operated on, by the minion.

terminalmage commented:
We allow this to be overridden in a remote-exec function, so enabling the same in orchestration seems like a logical extension of existing behavior.

twangboy committed:
Add _available_services function for testing

twangboy committed:
Fix lint, remove sentence from docstring

gtmanfred committed:
allow excluding multiple directories in rsync

Ch3LL committed:
Add for new source_* minion options

Ch3LL committed:
Change comment and salt.utils.network import

remyleone committed:
Declare set literally

MaximilianMeister committed:
fall back to PyMySQL

same is already done in modules (see #26803)

Signed-off-by: Maximilian Meister <mmeister@suse.de>

cbosdo committed:
virt.get_disks outputs the disk type

VM disks may also be floppies or cdroms. Output the libvirt device type
as a 'type' attribute of the disk for consumers to spot removable
devices (issue #46747).

cbosdo committed:
Add parameter to virt.purge to keep removable images.

Add a removables parameter to virt.purge to allow removing a VM without
removing the attached (and potentially shared) ISO removable images
(issue #46747).

boltronics committed:
Replace failing is and is not tests with == and !=

With the introduction of unicode_literals, these str to unicode
comparisons were incorrectly always returning False.

Fixes #46917.

terminalmage committed:
Add function to return TLS cert expiration date

Refs: #7424

sjorge committed:
Disable parsable output for zfs/zpool commands

Only very recent zfs on linux supports parsable output for zfs/zpool commands.
Also Oracle ZFS did not implement this at all.

Since we now do value normalization, we can deal with not having parsable output.

s0undt3ch committed:
Pass keyword arguments by name

s0undt3ch committed:
Advance `_next_fire_time` even when job didn't run

Sikerdebaard committed:
Added clone_images to OpenNebula template_clone kwargs. If set to True images attached to the template are cloned as well

Commit:
Merge pull request #46829 from beornf/handle-vault-exceptions

Exception handling in vault module

Commit:
Merge pull request #46794 from jdito/add-route-metrics

Adding the ability to specify a metric for static routes

Commit:
Merge pull request #46957 from cbosdo/libvirt_events_uuids

libvirt_events, harmonize object descriptions

pakdel committed:
Merge branch 'develop' into ec2_image_name

Commit:
Kill extra newline for lint

Commit:
Merge pull request #46941 from JamesSnow/develop

Fix at.atq support on FreeBSD. This also unbreaks at job creation with at.at.

Commit:
Tiny grammatical mistake in documentation

Commit:
Merge branch 'develop' into libvirt-events-fix

terminalmage committed:
Include exc_info in pillar render errors to aid in troubleshooting

terminalmage committed:
gpg renderer: fix tranceback when decrypted ciphertext contains binary data

terminalmage committed:
salt.payload.Serial: fix traceback when unpacking binary blob

dwoz committed:
Fix verify signature test

We load the key to memory first now so mocking salt.utils.fopen fixes
the signature verification test

gtmanfred committed:
support new versions of winrm-fs

Commit:
ssl -> SSL

Commit:
Merge pull request #46715 from pakdel/ec2_image_name

Salt Cloud EC2: Image Name

vtolstov committed:
returners/etcd: fix lint errors

Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

vtolstov committed:
Merge branch 'develop' into etcd

Commit:
Revert changes from #46520 that were not merged-forward correctly

This removes the changes from #46520 that were merged-forward and not handled
correctly in #46907.

Changes to `salt/utils/schedule.py` cannot be merged forward to develop as the
code there has be significantly refactored. The change in #46520 will need to
be applied directly to `develop` instead.

garethgreenaway committed:
Removing suse from pkgrepo comments tests.  the pkgrepo functions in SUSE pkg module do not support comments.

garethgreenaway committed:
Removing suse from pkgrepo comments tests.  the pkgrepo functions in SUSE pkg module do not support comments.

garethgreenaway committed:
Removing debugging.

garethgreenaway committed:
Removing debugging.

Commit:
Merge pull request #46389 from Sygnia/lvresize

linux_lvm module: refactor lvresize

Commit:
Merge pull request #46958 from brejoc/develop-fix-for-grain-test-core

Switch to assertCountEqual since order can change

Commit:
New "apply_to" kwarg in rabbitmq module should be added at the end

The change in #41233 added the "apply_to" kwarg to rabbitmq.set_policy. Since
the arg was not added at the end of the kwarg list, there was an API change.

In addition, when the order of "apply_to" and "priority" was swapped, the
corresponding kwargs were not changed, which causes a stack trace and the
rabbitmq_policy state to fail.

This change fixes #46880 so that the policy option will work correctly and we
can avoid this API change for the module. In 2017.7.x versions of Salt, the
order of args/kwargs is `vhost`, `name`, `pattern`, `definition`, `priority`,
and then `runas`. Therefore, `apply_to` must be last.

Fixes #46880

gtmanfred committed:
make sure windows outputs xml junit files

myinitialsarepm committed:
Update tests to use cmd.run_all

garethgreenaway committed:
Updating a couple tests.  Fixing check_whitelist_blacklist to work with PY3 when non-iterables are passed.  Adding warning about lst_avg results being wrong and future updates in Neon.

garethgreenaway committed:
Update test_pkgrepo.py

garethgreenaway committed:
Update test_pkgrepo.py

garethgreenaway committed:
Adding kwargs to calls into file module functions

Sikerdebaard committed:
Added documentation for clone_images var in OpenNebula template_clone function

Commit:
Fix unit tests for the checksum generator

Commit:
Use code checksum to update thin archive on client's cache

s0undt3ch committed:
There's no need to pass the job name just for logging purposes.

gtmanfred committed:
Merge pull request #46960 from Sikerdebaard/opennebula-template_clone-fix

Added clone_images to OpenNebula template_clone kwargs

Commit:
Lintfix

Commit:
Merge pull request #46970 from garethgreenaway/917_pkgrepo_comments_fix

 [2017.7] fix to pkgrepo comments test

Commit:
Merge branch '2018.3' into testing_jinja_filters_avaiable_via_salt_ssh

Commit:
Merge pull request #46944 from garethgreenaway/46881_Cron_file_source_file_not_found

[2018.3] cron.file with salt source URL

Commit:
Merge pull request #46967 from rallytime/fix-lint

Revert changes from #46520 that were not merged-forward correctly

Commit:
Merge pull request #46975 from gtmanfred/windows

Make windows work for test runs in jenkinsci

Commit:
Merge pull request #46955 from vtolstov/etcd

fixes for etcd returner/utils/sdb

Commit:
Merge pull request #46933 from sjorge/46903

Only very recent zfs support parsable output

Commit:
Merge pull request #46928 from cbosdo/libvirt-events-fix

Fix libvirt_events engine running on master

Commit:
Merge pull request #46888 from sieben/set_literal

Declare set literally

Commit:
Merge pull request #46692 from mattp-/develop

saltnado bugfixes for ldap & syndics

Commit:
Merge pull request #46657 from lvg01/fix-bug-46555

Changed destination for ipv6 in _ip_route_linux()

Commit:
Merge pull request #46966 from terminalmage/issue46934

Fix traceback when attempting to decode binary data to unicode

Ch3LL committed:
Fix redis cache log debug line

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - .kitchen.yml

Commit:
Fix: tuple instead of string

gtmanfred committed:
use saltstack salt-jenkins

Commit:
Merge branch '2018.3' into 'develop'

Conflicts:
  - salt/grains/core.py
  - salt/renderers/gpg.py

Commit:
Merge pull request #46991 from gtmanfred/windows

use saltstack salt-jenkins

Commit:
Merge branch '2017.7' into merge-2018.3

Conflicts:
  - .kitchen.yml

Commit:
Merge pull request #46973 from rallytime/fix-46880

New "apply_to" kwarg in rabbitmq module should be added at the end

Commit:
Merge pull request #46613 from myinitialsarepm/fix_puppet.fact_and_puppet.facts

Fix puppet.fact and puppet.facts to use stdout.

Commit:
Merge pull request #46848 from garethgreenaway/testing_jinja_filters_avaiable_via_salt_ssh

[2018.8] salt-ssh jinja filters tests

Commit:
Merge pull request #46989 from Ch3LL/redis_log

Fix redis cache log debug line

gtmanfred committed:
add unseal for vault runner

Ch3LL committed:
Patch ZMQ versions for master_uri test

Commit:
Merge pull request #46901 from cbosdo/purge_fix

Purge fix

Commit:
Merge pull request #46339 from DSRCorporation/bugs/ssh_state_test_failures

SSH State test failures

Commit:
Merge pull request #46990 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

Commit:
Merge pull request #46993 from L4rS6/fix-broken-keystone-auth/2018.3

Fix: tuple instead of string

Commit:
Merge branch '2017.7' into parallel-orch

Commit:
Merge pull request #46995 from s0undt3ch/develop

[schedule][develop] Minor fixes and enhancements

twangboy committed:
Fix unit test for win_ntp

LukeCarrier committed:
Fix respository (=> repository) typo in sls_build

This typo looks to have been introduced with 64aa4fb and would affect
anyone relying on the deprecated "name" attribute.

LukeCarrier committed:
Fix respository (=> repository) typo in sls_build

This typo looks to have been introduced with 64aa4fb and would affect
anyone relying on the deprecated "name" attribute.

Commit:
Merge pull request #46515 from gtmanfred/develop

allow excluding multiple directories in rsync

gtmanfred committed:
switch pip test package

Using irc3-plugins-test, since this should never be updated (i manage it and it
is fulfilling its only job).

LukeCarrier committed:
Add extra_filerefs to sls_build

Commit:
Merge pull request #46023 from bloomberg/parallel-orch

add parallel support for orchestrations

Commit:
Merge pull request #46526 from Ch3LL/ip_conf

Add tests for new source_* minion options

Commit:
Merge branch 'develop' into 45274-trafficserver-popen-arguments

twangboy committed:
Fix tests for Py3

readPlistFromString has been deprecated on Python3. Use loads instead.

lomeroe committed:
move exception for clarity

Commit:
Merge branch 'develop' into nested-orch-highstate

Commit:
Merge pull request #46999 from gtmanfred/2017.7

switch pip test package

ari committed:
Remove salt FreeBSD repo which is now 2 years out of date

The saltstack repository has the 2016 version is current. Its not being updated, but there is a perfectly good port in the FreeBSD tree to use instead. Enhance the documentation on how to install that from binary or source.

terminalmage committed:
Add saltenv to error message

garethgreenaway committed:
Fixing aliases in slack engine

garethgreenaway committed:
Fixing indention for slack documention.  Updating try..except to ensure we catch when groups aren't dicts.

terminalmage committed:
Rewrite file.patch state

This is a full rewrite. Previously, this state required the user to
provide the hash of the patched file, but this is no longer necessary as
the state is now able to check whether or not the patch will apply
cleanly without actually making any changes. The following is a summary
of the changes made in this rewrite:

- The `hash` and `dry_run_first` arguments are now ignored and generate
  a warning in the state return if used.
- The path being patched can now be a directory or a file, instead of
  just a file.
- The patch file is now downloaded by calling `file.managed`. This
  expands the potential file sources to include remote files (with all
  the hash verification features provided by `file.managed`), and also
  adds templating support.
- Instead of blindly running the `file.patch` remote-execution function
  and then checking that the patched result matches a pre-provided
  result, the state will run a non-dry-run `file.patch` with all patch
  output redirected to a temp file. This does 2 things: 1) It keeps any
  changes from being made, and 2) creates a reject file (which is not
  done when `--dry-run` is used with `patch`). This initial `file.patch`
  will tell us whether or not the patch would apply cleanly. If it does
  not, then we attempt to reverse-apply the rejected hunks using
  `--dry-run`, and if they apply cleanly then we know that the patch has
  already been applied and can exit with a `True` result. This means
  that we only actually attempt to apply the patch if we know if will
  apply cleanly, and prevents us from leaving the file(s) being patched
  in a partially-patched state, improving greatly the idempotence of the
  state.

terminalmage committed:
Rewrite file.patch state integration tests

This scraps the old tests and writes an entirely new test class with
over a dozen new tests, covering all the new functionality added in the
file.patch state rewrite.

terminalmage committed:
Add file.patch rewrite info to Fluorine release notes

jfoboss committed:
Merge pull request #1 from twangboy/pull_46539

Fix unit test for win_ntp

terminalmage committed:
Azure: ensure subscription_id is a str type

Resolves #46953.

brejoc committed:
Test is now also working with Python2

Since assertCountEqual was Python>=3.2 only, we backported this
to Python2.

opdude committed:
Don’t encode a unicode string

- We can’t encode once it’s already encoded

Commit:
Merge pull request #47012 from terminalmage/issue46953

Azure: ensure subscription_id is a str type

Commit:
Merge pull request #47014 from brejoc/develop-fix-for-grain-test-core

Test is now also working with Python2

Commit:
Merge pull request #46539 from jfoboss/patch-1

#46504 Fix

Commit:
Set master_top_first to False by default

Commit:
Merge pull request #45774 from twangboy/mac_add_service_util

Fix __virtual__ issue in mac_system.py

Commit:
Merge pull request #45313 from NTmatter/45274-trafficserver-popen-arguments

modules.trafficserver: Use lists for popen's commands and arguments

Commit:
Fix poor merge-conflict resolution: "translate_new_lines" doesn't exist in develop.

garethgreenaway committed:
Fixing test_path_which to check that the filter is available rather than results.

corywright committed:
Add auth.file module to auth documentation page

Commit:
Merge pull request #46997 from LukeCarrier/fix-docker-slsbuild-typo

Fix respository (=> repository) typo in sls_build

terminalmage committed:
Fix server_id grain in PY3 on Windows

https://github.com/saltstack/salt/pull/46649 caused a regression on
Windows since the executable path does not exist. This fixes that grain
on Windows by using `sys.executable`.

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - salt/state.py
  - salt/utils/files.py
  - tests/integration/modules/test_pip.py

Commit:
Update old fopen path with new utils files path

Commit:
Merge pull request #46994 from rallytime/merge-develop

[develop] Merge forward from 2018.3 to develop

Commit:
Merge pull request #46022 from bloomberg/nested-orch-highstate

salt.outputter.highstate: recursion for orch

Commit:
Merge pull request #46684 from isbm/isbm-saltssh-multimajor-py-version-support

SaltSSH cross-major version support

Commit:
Use msgpack utils for loads call, import msgpack for UnpackValueError

terminalmage committed:
Add to_str option to decode funcs

This allows for the string to be forced to a str type instead of
unicode on PY2.

terminalmage committed:
ldapmod.py/ldap3.py: Force modlist for search/modify/etc. to be str types

Commit:
Merge pull request #47022 from corywright/add-auth-file-module-to-docs

Add auth.file module to auth documentation page

gtmanfred committed:
Merge pull request #47020 from rallytime/bp-46970

Back-port #46970 to 2017.7

gtmanfred committed:
Merge pull request #47021 from garethgreenaway/920_state_module_jinja_filters_test_test_path_which

[2018.3] Fixing integration.modules.test_state_jinja_filters.StateModuleJinjaFiltersTest.test_path_which

Commit:
Merge pull request #47026 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

Commit:
Merge branch '2018.3' into 'develop'

Conflicts:
  - salt/cloud/clouds/azurearm.py

Commit:
Merge pull request #47023 from rallytime/bp-46997

Back-port #46997 to 2018.3

Commit:
Merge pull request #47027 from rallytime/bp-44508

Back-port #44508 to 2017.7

Commit:
Merge pull request #47025 from terminalmage/fix-server_id-windows

Fix server_id grain in PY3 on Windows

Commit:
Merge pull request #46913 from lomeroe/2017_7-fix46877

2017.7 Fix #46877 -- win_lgpo start/shutdown script reading

garethgreenaway committed:
Merge branch '2018.3' into crptodomekeyfix

Commit:
Merge pull request #47009 from garethgreenaway/46947_slack_documentation_update_catch_non_dicts

[2018.3] fixes to slack engine documentation

Commit:
Merge pull request #47008 from garethgreenaway/46951_fixing_slack_engine_aliases

[2018.3] Fixing aliases in slack engine

terminalmage committed:
Fix incorrect decorator path in salt/modules/lxd.py

NOTE: This bug only exists in develop

terminalmage committed:
Add ability to run a shell command to depends decorator

This adds the ability to suppress loading of a function based on the
retcode of a command.

Commit:
Merge pull request #46930 from dwoz/crptodomekeyfix

Clean up bad public key headers

Commit:
Merge pull request #47034 from rallytime/merge-develop

[develop] Merge forward from 2018.3 to develop

Commit:
Merge pull request #47035 from terminalmage/fix-decorator-path

Fix incorrect decorator path in salt/modules/lxd.py

Commit:
Merge pull request #47029 from terminalmage/issue46859

ldapmod.py/ldap3.py: Force modlist for search/modify/etc. to be str types

Commit:
Merge pull request #47019 from rallytime/bp-46919

Back-port #46919 to 2018.3

Commit:
Merge pull request #47005 from ari/patch-3

Remove salt FreeBSD repo which is now 2 years out of date

Commit:
Merge pull request #46996 from gtmanfred/vault

add unseal for vault runner

Commit:
Merge pull request #46892 from samilaine/update-doc-topics-event-events

Working Python examples for the event documentation.

rallytime commented:
@Tanoti There is a merge conflict on this PR. Can you rebase and fix the conflicts?

tankywoo committed:
Fix ba7d00f5 for gentoo pkg.installed

The commit ba7d00f5 change _cpv_to_cp inner implementation, but the
original parameter cpv is not changed, its format is `<p>-<v>`, and
`dep_getkey` need format for `=<p>-<v>` if `cpv` with version, so it cause
`InvalidAtom` exception and actually return cpv itself. So I add the
original implementation back to fix this problem and still keep
`dep_getkey` if the cpv format will be fixed in the future

clan committed:
switch order of file to be diffed

Orlando-Richards committed:
Return an empty dict instead of 'None' from grafana4 states

Fixes #46985

Orlando-Richards committed:
Remove accidentally added copy of a file

Commit:
Simplify LooseVersion check in `__virtual__` check in mac_assistive module

Commit:
Merge pull request #46890 from MaximilianMeister/pymysql-returner

fall back to PyMySQL

Commit:
Merge pull request #47017 from Unity-Technologies/hotfix/pip_windows

Don’t encode a unicode string

Commit:
Merge pull request #47048 from OrlandoArcapix/Issue46985-fix-grafana4-state

Issue46985 fix grafana4 state

Commit:
Merge pull request #47046 from clan/file_diff

switch order of file to be diffed

Commit:
Merge pull request #46326 from kstreee/fix-client-local

Fixes a timing bug of saltnado's client local.

clan committed:
handle jinja error in <module> level

prepare test case:
 # echo '{% import_yaml "a.yaml" as a with context %}' > a.sls
 # echo 'a: {{ "now"|strftime }}' > a.yaml

remove 'import salt.utils.daeutils' from salt/utils/templates.py, then run

 # salt-call state.sls a
[CRITICAL] Rendering SLS 'base:a' failed: Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 169, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 395, in render_jinja_tmpl
    line, out = _get_jinja_error(trace, context=decoded_context)
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 248, in _get_jinja_error
    line = _get_jinja_error_line(trace)
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 231, in _get_jinja_error_line
    return _get_jinja_error_slug(tb_data)[1]
TypeError: 'NoneType' object has no attribute '__getitem__'

after apply the patch:

 # salt-call state.sls a
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 169, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib64/python2.7/site-packages/salt/utils/templates.py", line 402, in render_jinja_tmpl
    tmplstr)
SaltRenderError: Jinja syntax error: no filter named 'strftime'
/var/cache/salt/minion/files/base/a.yaml(1):
---
a: {{ "now"|strftime }}    <======================
---
[CRITICAL] Rendering SLS 'base:a' failed: Jinja syntax error: no filter named 'strftime'
/var/cache/salt/minion/files/base/a.yaml(1):
---
a: {{ "now"|strftime }}    <======================
---

which shows accurate information for debug.

Commit:
Reduce the number of days an issue is stale by 10

corywright committed:
Fix copy/paste typo in minionfs tutorial

aabouzaid committed:
use log.error in case database is not defined

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - salt/modules/win_lgpo.py
  - salt/netapi/rest_tornado/saltnado.py

Commit:
Update old utils files to new new utils files path

terminalmage committed:
Parse all options before exiting

This allows for the state to report on multiple blacklisted opts

terminalmage committed:
Fix regression in handling of environment/saltenv

This was broken sometime between when the change was made and 2018.3.0
was released. This fixes the issue and adds a regression test.

Commit:
Merge pull request #47057 from corywright/fix-minionfs-whitelist-docs

Fix copy/paste typo in minionfs tutorial

Commit:
Merge pull request #47051 from rallytime/fix-lint

Simplify LooseVersion check in `__virtual__` check in mac_assistive module

Commit:
lint fix

beornf committed:
Support return value when setting vault secret

terminalmage committed:
Fix rst formatting

terminalmage committed:
Ignore all docstrings in runtests_decorators.py

terminalmage committed:
Remove unnecessary docstrings

beornf committed:
Add wrapped token auth method for vault module

sjorge committed:
Correct examples in smartos state docs to be valid yaml

aabouzaid committed:
add dbrp to define_task test

sjorge committed:
Ignore return code in zfs grains

When there are no pools zpool list exists with 1.
It's possible and OK to have zfs but no have any pools.

```
[ERROR   ] Command '/usr/sbin/zpool list -H -o name,size' failed with return code: 1
[ERROR   ] output:
```

awerner committed:
saltclass: added node/class pillars to __pillar__

Add found pillars to the __pillar__ variable when encountered.
This allows referencing other __pillar__ dynamically and also more
complex jinja operations referencing in saltclass defined pillars.

mattp- committed:
GH-26396, add a clarification on pillar merge strategy

make it clear what is required for yamlex behavior cross-sls.

ixs committed:
Add Xen function to get and set PV_ARGS

PV_ARGS are used to pass additional commandline arguments to the
kernel when running XenPV domains.

The new actions can be run as follows:

salt-cloud -a get_pv_args xenvm01
salt-cloud -a set_pv_args xenvm01 pv_args="utf-8 graphical"

athienem committed:
Minor spelling fix: templat -> template

campbellmc committed:
Update consul.py

Added  param comment regarding ignoring data for GET requests in _query.

remyleone committed:
Add Scaleway to cloud providers examples

Commit:
Merge pull request #47099 from sieben/cloud_providers_salt

Add Scaleway to cloud providers examples

Commit:
Merge branch 'develop' into zfsgrainnopool

remyleone committed:
Use internal http.query library for HTTP requests

terminalmage committed:
Add regex matches to expr_match unit tests

Commit:
Merge pull request #47075 from sjorge/docfixvmadm

Correct examples in smartos state docs to be valid yaml

Commit:
Merge branch 'develop' into vault-wrapped-token

Commit:
Merge branch 'develop' into vault-write-data

Commit:
Merge pull request #47066 from terminalmage/issue46979

Fix regression in handling of environment/saltenv

Commit:
Merge pull request #47062 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

garethgreenaway committed:
Fixing a bug that occured when a comment was added to a message sent to Slack by Salt.  Also making `slack_engine:groups_pillar` optional.

Commit:
Merge pull request #47054 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 10

Commit:
Merge pull request #47053 from clan/jinja-error

handle jinja error in <module> level

Commit:
Merge pull request #47010 from terminalmage/issue40146

Rewrite file.patch state

Commit:
Merge pull request #46825 from AAbouZaid/kapacitor_enhancements

Follow up Kapacitor module enhancements.

terminalmage committed:
Update default renderer to jinja|yaml, improve renderer docs

`yaml_jinja` and all legacy renderer syntax has been transparently
translated to the render pipeline syntax (e.g `jinja|yaml`)
under-the-hood for several years now, this just makes the default
renderer adhere to the render pipeline syntax.

It also makes significant improvements to the renderer documentation.

Commit:
salt-cloud/vmware if specified use ssh_host to deploy

Fixes #36543

Commit:
fix pip.req import error in pip 10.0.0

Commit:
Merge pull request #47101 from sieben/fix_scaleway

Use internal http.query library for HTTP requests

Commit:
Merge pull request #45970 from dxclabs/consul_remove_data_for_GET_requests

stop 'data' being assigned with serialised empty dict for GET requests

Commit:
Merge pull request #47045 from tankywoo/fix-gentoo-pkg-installed

Fix ba7d00f5 for gentoo pkg.installed method

Commit:
Merge pull request #47109 from garethgreenaway/46943_slack_engine_fixes

[2018.3] fixes to Slack engine

Commit:
Merge branch 'develop' into vault-write-data

Commit:
Merge pull request #47105 from terminalmage/issue46768

Add regex matches to expr_match unit tests

Commit:
add ImportError

Commit:
Merge branch 'develop' into zfsgrainnopool

Commit:
Merge pull request #47082 from awerner/add-saltclass-recursive-pillars

saltclass: added pillars to __pillar__ when encountered

terminalmage committed:
Update test to reflect docstring changes

Commit:
Merge branch '2018.3' into 'develop'

Conflicts:
  - salt/netapi/rest_tornado/saltnado.py

Commit:
Update old utils paths with new utils paths

Commit:
Merge pull request #47121 from pcjeff/develop

fix pip import error in pip 10.0.0

Commit:
Merge pull request #47123 from rallytime/merge-develop

[develop] Merge forward from 2018.3 to develop

Commit:
Merge pull request #47111 from terminalmage/default-renderer

Update default renderer to jinja|yaml, improve renderer docs

Commit:
Merge pull request #47091 from ixs/xen_pv_args

Add Xen function to get and set PV_ARGS

Commit:
Merge pull request #47078 from sjorge/zfsgrainnopool

Ignore return code in zfs grains

Commit:
Merge pull request #47072 from beornf/vault-wrapped-token

Add wrapped token auth method for vault module

Commit:
Merge pull request #47071 from beornf/vault-write-data

Support return value when setting vault secret

Commit:
Merge pull request #47084 from bloomberg/26396-docpatch

GH-26396, add a clarification on pillar merge strategy

dijit committed:
plaintext not supported in Windows Server 2016.

ntlm authentication is backwards compatible to Windows Server 2008 at least, but plaintext is not supported in Server 2016.

Commit:
only change ssh_host if necessary

Commit:
Merge pull request #47115 from cvedel/allow-ssh-host-in-vm-profile

salt-cloud/vmware if specified use ssh_host to deploy

Commit:
Merge pull request #47138 from dijit/patch-1

plaintext not supported in Windows Server 2016.

Commit:
Change check_db_exists default to True

This option was added in PR #46895 where the default in the `run_file` function
was set to `True`, but in the `run` function it was set to `False`. This is an
error and the default should be `True`.

With the default set to False, it causes a test failure and would also be a change
in behavior from previous releases. In addition, the docs also say that default
is True. Therefore, this looks to be a typo. This PR sets the default to `True`.

Fixes the following test failure:

- unit.states.test_mysql_query.MysqlQueryTestCase.test_run

terminalmage committed:
Remove trailing whitespace

terminalmage committed:
mocked file_roots and pillar_roots should be dicts

terminalmage committed:
Mock __gather_avail in test_includes_override_sls

Commit:
Merge pull request #47148 from LukeCarrier/add-docker-slsbuild-extrafilerefs

Add extra_filerefs to sls_build

terminalmage committed:
Log the exception when the read/cert object creation fails

terminalmage committed:
Add unit test for tls.get_expiration_date

Commit:
Merge pull request #47161 from terminalmage/salt-jenkins-930

Fix failing pillar unit test

Commit:
Merge pull request #47040 from terminalmage/new-decorators

Add ability to run a shell command to depends decorator

Commit:
Merge pull request #47158 from rallytime/fix-mysql-test

Change check_db_exists default to True

Commit:
Merge pull request #46922 from terminalmage/issue7424

Add function to return TLS cert expiration date

Tanoti committed:
Merge branch 'pass-test-mode-in-orchestration' of https://github.com/Tanoti/salt into pass-test-mode-in-orchestration

rallytime reviewed (changes requested):
@Tanoti Something has gone awry here. Can you rebase this?

Tanoti commented:
@rallytime I thought I had rebased it as I certainly ran the commands on my branch. I'm not an experienced git user so it possible I made a mistake. What do I need to do?

rallytime commented:
@Tanoti There's some info in step 6 of [these docs](https://docs.saltstack.com/en/latest/topics/development/contributing.html#sending-a-github-pull-request)

Let us know if you have questions!

rallytime commented:
@Tanoti Just checking in here. Were you able to resolve the issues here? There are merge conflicts on this PR now as well. Feel free to close this and open a new PR if that is easier.

Tanoti commented:
@rallytime I think I'll have to open a new PR as I've gotten this one in a bit of a mess.

rallytime commented:
Great! Thank you!
